ktor-server/ktor-server-core/src/io/ktor/features/ContentNegotiation.kt

```diff
@@ -73,32 +75,24 @@ class ContentNegotiation(val converters: List<ContentConverterRegistration>) {
             }
             return feature
         }
-
-        private fun PipelineContext<Any, ApplicationCall>.acceptedTypes(): List<Pair<ContentType, Double>> {
-            val acceptHeader = call.request.header(HttpHeaders.Accept)
-            val acceptHeaderValues = parseHeaderValue(acceptHeader)
-            return acceptHeaderValues
-                    .map { ContentType.parse(it.value) to it.quality }
-                    .sortedWith(contentTypeComparator)
-        }
     }
 }
 
 class UnsupportedMediaTypeException(contentType: ContentType) : Exception("Content type $contentType is not supported")
 
 interface ContentConverter {
-    suspend fun convertForSend(context: PipelineContext<Any, ApplicationCall>, value: Any): Any?
+    suspend fun convertForSend(context: PipelineContext<Any, ApplicationCall>, contentType: ContentType, value: Any): Any?
     suspend fun convertForReceive(context: PipelineContext<ApplicationReceiveRequest, ApplicationCall>): Any?
 }
 
-class ConvertedContent(text: String, val contentType: ContentType) : OutgoingContent.ByteArrayContent() {
-    private val bytes = text.toByteArray(Charsets.UTF_8)
+class ConvertedContent(val text: String, val contentType: ContentType, override val status: HttpStatusCode? = null) : OutgoingContent.ByteArrayContent() {
+    private val bytes by lazy { text.toByteArray(contentType.charset() ?: Charsets.UTF_8) }
```

Comment:
text will be converted to bytes anyway (unless gets filtered out) so I see no reason to use lazy delegation here as it costs few more object instances

Reply:
It's almost direct copy of TextContent. Although I don't see any reason for introducing new OutgoingContent classes, especially tuned for text-only representation. May be better solution is completely remove ConvertedContent and delegate choosing correct classes to converters?